CONTRIBUTING.md

```diff
@@ -87,31 +87,31 @@ The dependency-version should be set to the most recent released version and the
 
 `com.azure:azure-identity;1.0.0-preview.4;1.0.0-preview.5`
 
-Note: In the case of a new or unreleased artifact both versions will be the same.
+Note: In the case of a new artifact both versions will be the same. In the case of a released artifact, the dependecny version should be the latest released version.
 
 ### Libraries vs External Dependencies
 
 Libraries refer to things that are built and released as part of the Azure SDK. Libraries have a current version and a dependency version.
 
-External Dependencies refer to dependencies for things that are not built and released as part of the Azure SDK regardless of the source. External Dependencies will only ever have one version.
+External Dependencies refer to dependencies for things that are not built and released as part of the Azure SDK regardless of the source. External Dependencies will only ever have a dependency version.
 
-### Current version vs Dependency version
+### Current version, Dependency version and Unreleased Dependency version
 
 Current version - This is the version we should be using when defining a component in its POM file and also when dependent components are built within the same pipeline. The current version is the version currently in development.
-Dependency version - This is the version we should be using when a given library is a dependency outside of a particular area. This should be the latest released version of the package.
+Dependency version - This is the version we should be using when a given library is a dependency outside of a particular area. This should be the latest released version of the package whenever possible.
+Unreleased Dependency version – Whenever possible, libraries should be using the latest released version for dependencies but there is the case where active development in one library is going to be needed by another library or libraries that are built in separate pipelines. Typically these changes are additive, not breaking, and the reason for this is that once a library is GA’d, nothing short of breaking changes should ever force the dependency versions across the repo to an unreleased version as this would prevent other libraries, that don’t need this change, from releasing.
```

Comment:
The last line, ` Typically these changes are additive ….` ,  is very long and with many punctuations. Although it is accurate technically but little bit difficult to understand. Request to word it in smaller sentences for better readability.

Reply:
Updated with the second PR. Broke the massive run-on into several smaller sentences.